infra/resources/prod/function_profile.tf

```
@@ -0,0 +1,230 @@
+      #
+
+      # UNIQUE EMAIL ENFORCEMENT
+      // TODO: searching the following 2 env config in the codebase show no results
+      FF_UNIQUE_EMAIL_ENFORCEMENT    = "ALL"
+      UNIQUE_EMAIL_ENFORCEMENT_USERS = jsonencode(split(",", data.azurerm_key_vault_secret.app_backend_UNIQUE_EMAIL_ENFORCEMENT_USER.value))
+
+      PROFILE_EMAIL_STORAGE_CONNECTION_STRING = data.azurerm_storage_account.citizen_auth_common.primary_connection_string
+      PROFILE_EMAIL_STORAGE_TABLE_NAME        = "profileEmails"
+      ON_PROFILE_UPDATE_LEASES_PREFIX         = "OnProfileUpdateLeasesPrefix-001"
+
+      MAILUP_USERNAME = data.azurerm_key_vault_secret.common_MAILUP_USERNAME.value
+      MAILUP_SECRET   = data.azurerm_key_vault_secret.common_MAILUP_SECRET.value
+      PUBLIC_API_KEY  = trimspace(data.azurerm_key_vault_secret.fn_app_PUBLIC_API_KEY.value)
+      // TODO: probably does not needed anymore due the fn which was using it is now moved in io-profile-async
+      SPID_LOGS_PUBLIC_KEY = trimspace(data.azurerm_key_vault_secret.fn_app_SPID_LOGS_PUBLIC_KEY.value)
+      AZURE_NH_ENDPOINT    = data.azurerm_key_vault_secret.fn_app_AZURE_NH_ENDPOINT.value // TODO: searching this config in the codebase show no results
+
+    }
+  }
+}
+
+
+module "function_profile" {
+  source  = "pagopa-dx/azure-function-app/azurerm"
+  version = "~> 0.0"
+
+  environment = {
+    prefix          = local.prefix
+    env_short       = local.env_short
+    location        = local.location
+    domain          = local.domain
+    app_name        = local.function_profile.name
+    instance_number = "02"
+  }
+
+  node_version          = 20
+  health_check_path     = "/api/v1/info"
+  has_durable_functions = "true"
+
+  # P2mv3 SKU and 8 Worker process count(sku set as of io-infra function-profile definition)
+  tier = "xl"
+
+
+  resource_group_name = azurerm_resource_group.main_resource_group.name
+
+  subnet_cidr   = local.cidr_subnet_fn_profile
+  subnet_pep_id = data.azurerm_subnet.private_endpoints_subnet.id
+
+  private_dns_zone_resource_group_name = data.azurerm_resource_group.rg_common.name
+
+  virtual_network = {
+    name                = data.azurerm_virtual_network.itn_common.name
+    resource_group_name = data.azurerm_virtual_network.itn_common.resource_group_name
+  }
+
+  app_settings      = local.function_profile.app_settings
+  slot_app_settings = local.function_profile.app_settings
+
+  subnet_service_endpoints = {
+    web = true
+  }
+
+  application_insights_connection_string = data.azurerm_application_insights.application_insights.connection_string
+
+  action_group_id = azurerm_monitor_action_group.error_action_group.id
+
+  tags = local.tags
+}
+
+module "function_profile_autoscale" {
+  depends_on = [module.function_profile]
+  source     = "pagopa/dx-azure-app-service-plan-autoscaler/azurerm"
+  // TODO: in order to update to version 1.0.0, add the required inputs `app_service_plan_id` and `location`
+  version = "0.0.2"
```

suggestion: what about

```suggestion
  version = "~> 0.0"
```

this will incorporate all _patch_ and _minors_ update (`0.x`)